api/namex/resources/requests.py

```diff
@@ -749,6 +749,17 @@ def put(nr, *args, **kwargs):
                 json_input['submittedDate'] = str(datetime.strptime(
                     str(json_input['submittedDate'][5:]), '%d %b %Y %H:%M:%S %Z'))
 
+            # Convert Expiration Date to the correct format
+            if json_input.get('expirationDate', None):
+                expiration_str = json_input['expirationDate']
+                if expiration_str.endswith('UTC'):
+                    parsed_date = datetime.strptime(expiration_str[5:], '%d %b %Y %H:%M:%S %Z')
+                else:
+                    parsed_date = datetime.fromisoformat(expiration_str.rstrip('Z'))
+
+                end_of_day = parsed_date.replace(hour=23, minute=59, second=59)
+                json_input['expirationDate'] = end_of_day.strftime('%Y-%m-%d %H:%M:%S')
+
             # convert NWPTA dates to correct format
```

Comment:
I recall that the NR expiration date should be set to the end of the day in Pacific Time. Please verify to confirm

Reply:
You are right @eve-git, I missed this part. I just updated it so that when expiration dates are updated they are set to the end of day pacific time. I added checks so that it would work with both UTC and ISO formats.

I tested this locally and now when I update the expiration date the time stays as `YYYY-MM-DD 06:59:00+00` representing the end of day pacific time in UTC format. I went off matching the format to how the expiration date is initially created in `create_expiry_date` https://github.com/bcgov/namex/blob/main/api/namex/services/name_request/abstract_name_request.py

This is keeping expiry dates consistent for me locally. Another option to fix this would be to revert the two PRs I mentioned above so the date stays in UTC format and does not get converted to ISO in the front end. That would require 2 PRs though but would prevent having to handle two different time formats.